datafusion/datasource-parquet/src/file_format.rs

```diff
@@ -453,29 +451,8 @@ impl FileFormat for ParquetFormat {
         Ok(Arc::new(DataSinkExec::new(input, sink, order_requirements)) as _)
     }
 
-    fn supports_filters_pushdown(
-        &self,
-        file_schema: &Schema,
-        table_schema: &Schema,
-        filters: &[&Expr],
-    ) -> Result<FilePushdownSupport> {
-        if !self.options().global.pushdown_filters {
-            return Ok(FilePushdownSupport::NoSupport);
-        }
-
-        let all_supported = filters.iter().all(|filter| {
-            can_expr_be_pushed_down_with_schemas(filter, file_schema, table_schema)
-        });
-
-        Ok(if all_supported {
-            FilePushdownSupport::Supported
-        } else {
-            FilePushdownSupport::NotSupportedForFilter
-        })
-    }
-
     fn file_source(&self) -> Arc<dyn FileSource> {
-        Arc::new(ParquetSource::default())
+        todo!() // need access of file schema?
```

Comment:
This poses an issue.

TLDR is that in order to know if it can absorb a filter as exact `ParquetSource` needs to know not only the filter but also the file schema it's applied to (in particular to get the type of the columns since it can't handle structs).